rust/shalom_dart_codegen/dart_tests/test/custom_scalar_arguments/test.dart

```diff
@@ -0,0 +1,147 @@
+
+  test("required custom scalar argument", () {
+    var variables = UpdatePointCoordsNonNullVariables(coords: samplePoint);
+    expect(variables.toJson(), {"coords": samplePointRaw});
+    var variablesUpdated = variables.updateWith(coords: updatedPoint);
+    expect(variablesUpdated.coords, updatedPoint);
+    final req =
+        RequestUpdatePointCoordsNonNull(
+          variables: UpdatePointCoordsNonNullVariables(coords: updatedPoint),
+        ).toRequest();
+    expect(req.variables, {"coords": updatedPointRaw});
+  });
+
+  test("optional custom scalar argument", () {
+    var variables = UpdatePointCoordsMaybeVariables(coords: Some(samplePoint));
+    var variablesUpdated = variables.updateWith(
+      coords: Some(Some(updatedPoint)),
+    );
+    expect(variablesUpdated.coords.some(), updatedPoint);
+    expect(
+      RequestUpdatePointCoordsMaybe(
+        variables: UpdatePointCoordsMaybeVariables(coords: Some(null)),
+      ).toRequest().variables,
+      {"coords": null},
+    );
+
+    expect(
+      RequestUpdatePointCoordsMaybe(
+        variables: UpdatePointCoordsMaybeVariables(coords: Some(samplePoint)),
+      ).toRequest().variables,
+      {"coords": samplePointRaw},
+    );
+    expect(
+      RequestUpdatePointCoordsMaybe(
+        variables: UpdatePointCoordsMaybeVariables(coords: None()),
+      ).toRequest().variables,
+      {},
+    );
+  });
+
+  test("required custom scalar argument in InputObject", () {
+    final variables = UpdatePointWithInputNonNullVariables(
+      pointData: PointDataInput(coords: samplePoint, name: "Location A"),
+    );
+    final variablesUpdated = variables.updateWith(
+      pointData: PointDataInput(coords: updatedPoint, name: "Location B"),
+    );
+    expect(variablesUpdated.pointData.coords, updatedPoint);
+
+    final req =
+        RequestUpdatePointWithInputNonNull(
+          variables: UpdatePointWithInputNonNullVariables(
+            pointData: PointDataInput(
+              coords: Point(x: 50, y: 60),
+              name: "Location C",
+            ),
+          ),
+        ).toRequest();
+    expect(req.variables, {
+      "pointData": {"coords": "POINT (50, 60)", "name": "Location C"},
+    });
+  });
+
+  test("optional custom scalar argument in InputObject", () {
+    final variables = UpdatePointWithInputCoordsMaybeVariables(
+      pointData: PointUpdateCoordsMaybe(coords: Some(null), name: "Location D"),
+    );
+    final variablesUpdated = variables.updateWith(
+      pointData: PointUpdateCoordsMaybe(
+        coords: Some(updatedPoint),
+        name: "Location E",
+      ),
+    );
+    expect(variablesUpdated.pointData.coords, Some<Point?>(updatedPoint));
+    final req =
+        RequestUpdatePointWithInputCoordsMaybe(
+          variables: variables,
+        ).toRequest();
+    expect(req.variables, {
+      "pointData": {"coords": null, "name": "Location D"},
+    });
+  });
+
+  test("optional custom scalar argument with default value", () {
```

Comment:
```suggestion
  test("custom scalar argument with (optional default value)", () {
```

Reply:
this test should be upwards, ATMOF you can put all the input object test inside a test group